### PR TITLE
vlan:fix cannot use "Set-NetAdapter" in powershell on windows7 guest

### PR DESCRIPTION
**Pull request by kimi1978wy**
Win7 and win2008 cannot use "Set-NetAdapter" in powershell, so replace with "Netsh".

Signed-off-by: Yu Wang <wyu@redhat.com>

id:1657625

### CODE REVIEW

**Comment by kimi1978wy** [comment]
(1/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.vlan.q35: PASS (222.82 s)
 (2/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.i386.sp1.io-github-autotest-qemu.vlan.q35: PASS (115.65 s)
 (3/3) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.vlan.q35: PASS (135.74 s)
 (1/4) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.vlan.q35: PASS (311.65 s)
 (2/4) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.vlan.q35: PASS (139.90 s)
 (3/4) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.vlan.q35: PASS (176.74 s)
 (4/4) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.1.io-github-autotest-qemu.vlan.q35: PASS (158.35 s)

File: generic/tests/vlan.py [hunk 1]
```diff
@@ -141,6 +143,34 @@ def flood_ping(src, dst):
...
+        Get the proper netkvmco.dll path from iso.
+
+        :param session: a session to send cmd
+        :return: the proper netkvmco.dll path
+        """
+
+        viowin_ltr = virtio_win.drive_letter_iso(session)
+        if not viowin_ltr:
+            err = "Could not find virtio-win drive in guest"
+            raise exceptions.TestError(err)
```
Comment (ybduan) [inline generic/tests/vlan.py]: If I remember correctly, it is recommended to use test.error() here.

**Comment by kimi1978wy** [comment]
@yduan updated , pls help to review it again
@pezhang Could you help to review this patch? Thanks a lot

**Comment by kimi1978wy** [comment]
(01/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.ctrl_vlan.ctrl_on.q35: PASS (150.21 s)
 (02/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.ctrl_vlan.ctrl_off.q35: PASS (151.95 s)
 (05/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.ctrl_vlan.ctrl_on.q35: PASS (74.76 s)
 (06/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win7.x86_64.sp1.io-github-autotest-qemu.ctrl_vlan.ctrl_off.q35: PASS (54.08 s)
 (07/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.ctrl_vlan.ctrl_on.q35: PASS (101.32 s)
 (08/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.io-github-autotest-qemu.ctrl_vlan.ctrl_off.q35: PASS (74.88 s)
 (09/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.1.io-github-autotest-qemu.ctrl_vlan.ctrl_on.q35: PASS (97.46 s)
 (10/10) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.1.io-github-autotest-qemu.ctrl_vlan.ctrl_off.q35: PASS (73.26 s)

 (1/1) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.ctrl_vlan.ctrl_on.q35: PASS (127.64 s)
 (1/1) Host_RHEL.m8.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2008.x86_64.r2.io-github-autotest-qemu.ctrl_vlan.ctrl_on.q35: PASS (147.29 s)

File: qemu/tests/ctrl_vlan.py [hunk 2]
```diff
@@ -38,41 +41,89 @@ def verify_vlan_table(expect_vlan=None):
...
+        Get the proper netkvmco.dll path from iso.
+
+        :param session: a session to send cmd
+        :return: the proper netkvmco.dll path
+        """
+
+        viowin_ltr = virtio_win.drive_letter_iso(session)
+        if not viowin_ltr:
+            err = "Could not find virtio-win drive in guest"
+            raise exceptions.TestError(err)
```
Comment (hereischen) [inline qemu/tests/ctrl_vlan.py]: test.errer()

File: generic/tests/vlan.py [hunk 3]
```diff
@@ -141,6 +142,34 @@ def flood_ping(src, dst):
...
+            err = "Could not get product dirname of the vm"
+            test.error(err)
+        guest_arch = virtio_win.arch_dirname_iso(session)
+        if not guest_arch:
+            err = "Could not get architecture dirname of the vm"
+            test.error(err)
+
+        middle_path = "%s\\%s" % (guest_name, guest_arch)
+        find_cmd = 'dir /b /s %s\\netkvmco.dll | findstr "\\%s\\\\"'
+        find_cmd %= (viowin_ltr,  middle_path)
```
Comment (hereischen) [inline generic/tests/vlan.py]: If I were writing this, I will just format once.
`find_cmd = dir /b /s %s\\netkvmco.dll | findstr "\\%s\\%s\\\\" % (viowin_ltr, guest_name, guest_arch)`
But you can keep it this way

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

**Comment by pezhang** [comment]
LGTM.

Acked by: Pei Zhang pezhang@redhat.com

**Pull request merged**